core/src/main/scala/kafka/server/metadata/MetadataPartitions.scala

```diff
@@ -30,7 +31,8 @@ import scala.jdk.CollectionConverters._
 
 
 object MetadataPartition {
-  def apply(name: String, record: PartitionRecord): MetadataPartition = {
+  val OffsetNeverDeferred = 0L // must not be a valid offset we could see (i.e. must not be positive)
```

Comment:
Can you add JavaDoc for this?

Reply:
As discussed offline, we will remove this and not include the last seen offset in log messages when applying deferred changes.